### ISSUE TITLE
Bug: Accessibility > Ensure controls are not nested

### ISSUE THREAD

**Issue by ogi-m** [mattermost/focalboard#2227]
Ensure controls are not nested as they are not announced by screen readers ([more info](https://dequeuniversity.com/rules/axe/4.3/nested-interactive?application=AxeChrome)).

1. Sidebar 1
    - Element location: 
 `.\' > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
`<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button IconButton"><i class="CompassIcon icon-dots-horizontal OptionsIcon"></i></button></div>`
    
2. Header 1
    - Element location: 
 `.ViewHeader > .MenuWrapper[role="button"][aria-label="menuwrapper"]:nth-child(2)`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button IconButton"><i class="CompassIcon icon-chevron-down DropdownIcon"></i></button></div>`

3. Header 2
    - Element location: 
 `.ViewHeader > .MenuWrapper[role="button"][aria-label="menuwrapper"]:nth-child(4)`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button"><span>Properties</span></button></div>`
4. Header 3
    - Element location: 
 `.MenuWrapper[role="button"][aria-label="menuwrapper"]:nth-child(5)`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button"><span>Group by: <span id="groupByLabel" style="color: rgb(var(--center-channel-color-rgb));">Type</span></span></button></div>`
5. Header 4
    - Element location: 
 `.MenuWrapper[role="button"][aria-label="menuwrapper"]:nth-child(7)`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button"><span>Sort</span></button></div>`
6. Header 5
    - Element location: 
 `.ModalWrapper:nth-child(9) > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button IconButton"><i class="CompassIcon icon-dots-horizontal OptionsIcon"></i></button></div>`

7. Board 1
    - Element location: 
 `.KanbanColumnHeader.octo-board-header-cell[draggable="true"]:nth-child(1) > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
   `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button IconButton"><i class="CompassIcon icon-dots-horizontal OptionsIcon"></i></button></div>`

8. Card 1
    - Element location: 
 `.toolbar > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button IconButton IconButton--large"><i class="CompassIcon icon-dots-horizontal OptionsIcon"></i></button></div>`
9. Card 2
    - Element location: 
 `.octo-propertyrow:nth-child(1) > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><div class="octo-propertyname"><button type="button" class="Button"><span>Type</span></button></div></div>`
10. Card 3
    - Element location: 
 `.add-property > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button"><span>+ Add a property</span></button></div>`
11. Card 4
    - Element location: 
 `.CardDetailContentsMenu > .MenuWrapper[role="button"][aria-label="menuwrapper"]`
    - Element source: 
    `<div role="button" aria-label="menuwrapper" class="MenuWrapper"><button type="button" class="Button"><span>Add content</span></button></div>`
